addons/sale/models/sale.py

```diff
@@ -166,8 +163,6 @@ def _compute_tax_id(self):
     signature = fields.Binary('Signature', help='Signature received through the portal.', copy=False, attachment=True)
     signed_by = fields.Char('Signed by', help='Name of the person that signed the SO.', copy=False)
 
-    product_id = fields.Many2one('product.product', related='order_line.product_id', string='Product')
```

Comment:
You don't like this field ? 

More seriously, are you sure that you don't have any side effect by removing it ?

This field is not stored, so it's not a big deal to keep it if we are not sure that there won't be any traceback because product_id is a PITA to grep ;)

Reply:
Hello @rco-odoo 
We already discussed it in person, but can you confirm a related field of a many2one through a one2many relation is indeed something we should remove?

The only 2 uses I have found are: 

- used in a search, which I fixed
- used in the ebay module (see enterprise PR)

If we remove it, I'll probably do it in a separate commit then.

Reply:
I reviewed it with ODO and AL and we decided it was ok to remove it, since it is directly impacted by the fact product_id is not required anymore.

In practice used in a search the behaviour will be that of "any" which was actually fine and intended, but there is a another way to do the filter so it's ok to remove the field.

Then the field used by itself does "first" which could cause issues if the first line is expected to have product, so we remove it to be safe. The (supposedly) only case where it was used even tried incorrectly to use it as "any" instead of "first".